traffic_ops/traffic_ops_golang/server/servers_update_status.go

```diff
@@ -46,18 +46,148 @@ func GetServerUpdateStatusHandler(w http.ResponseWriter, r *http.Request) {
+	LEFT JOIN status AS pstatus ON pstatus.id = ps.status
+	WHERE pstatus.name != 'OFFLINE'
+), use_reval_pending AS (
+	SELECT value::BOOLEAN
+	FROM parameter
+	WHERE name = 'use_reval_pending'
+	AND config_file = 'global'
+	UNION ALL SELECT FALSE FETCH FIRST 1 ROW ONLY
+)
+SELECT
+	s.id,
+	s.host_name,
+	type.name AS type,
+	(s.reval_pending::BOOLEAN) AS server_reval_pending,
+	use_reval_pending.value,
+	s.upd_pending,
+	status.name AS status,
+	COALESCE(BOOL_OR(ps.upd_pending),
+		TRUE IN (
+			SELECT sta.upd_pending FROM  server_topology_ancestors sta)
+		) AS parent_upd_pending,
+	COALESCE(BOOL_OR(ps.reval_pending),
+		TRUE IN (
+			SELECT sta.reval_pending FROM  server_topology_ancestors sta)
+		) AS parent_reval_pending
+	FROM use_reval_pending,
+		 server s
+LEFT JOIN status ON s.status = status.id
+LEFT JOIN cachegroup cg ON s.cachegroup = cg.id
+LEFT JOIN type ON type.id = s.type
+LEFT JOIN parentservers ps ON ps.cachegroup = cg.parent_cachegroup_id
+	AND ps.cdn_id = s.cdn_id
+	AND type.name = 'EDGE'
+WHERE s.host_name = $1
+GROUP BY s.id, s.host_name, type.name, server_reval_pending, use_reval_pending.value, s.upd_pending, status.name
+ORDER BY s.id
+` // remove the type.name = 'EDGE' condition if other server types should have their parents processed
+
+	rows, err = tx.Query(selectQuery, hostName)
+	if err != nil {
+		log.Error.Printf("could not execute query: %s\n", err)
+		return nil, tc.DBError
+	}
+	defer log.Close(rows, "unable to close db connection")
+
+	for rows.Next() {
+		var serverUpdateStatus tc.ServerUpdateStatus
+		var serverType string
+		if err := rows.Scan(&serverUpdateStatus.HostId, &serverUpdateStatus.HostName, &serverType, &serverUpdateStatus.RevalPending, &serverUpdateStatus.UseRevalPending, &serverUpdateStatus.UpdatePending, &serverUpdateStatus.Status, &serverUpdateStatus.ParentPending, &serverUpdateStatus.ParentRevalPending); err != nil {
+			log.Error.Printf("could not scan server update status: %s\n", err)
+			return nil, tc.DBError
+		}
+		updateStatuses = append(updateStatuses, serverUpdateStatus)
+	}
+	return updateStatuses, nil
+}
+
+func GetServerUpdateStatusHandlerV2(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"host_name"}, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	serverUpdateStatus, err := getServerUpdateStatusV2(inf.Tx.Tx, inf.Config, inf.Params["host_name"])
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+		return
+	}
+	api.WriteRespRaw(w, r, serverUpdateStatus)
+}
+
+// getServerUpdateStatusV2 supports /servers/all/update_status in addition to /servers/{host_name}/update_status
+// This special case is believed to be used nowhere.
+func getServerUpdateStatusV2(tx *sql.Tx, cfg *config.Config, hostName string) ([]tc.ServerUpdateStatus, error) {
+	// language=SQL
+	baseSelectStatement := `
```

Comment:
Nitpicks: Complex SQL could use comments, and magic strings would be better as constants (OFFLINE, use_reval_pending, global)

Reply:
>Complex SQL could use comments

* Commented query additions in f97f121267, this one is the legacy query.

> magic strings would be better as constants (OFFLINE, use_reval_pending, global)

* Using `tc.CacheStatusOffline` instead of 'OFFLINE' in 5ef79b21cb

* Using `tc.UseRevalPendingParameterName` instead of `'use_reval_pending'` ccb41948c2

* Use `tc.GlobalConfigFileName` instead of hard-coded `'global'` in b1e912424d